FoodplannerDataAccessSql/Codes/OneTimePasswordRepository.cs

```diff
@@ -0,0 +1,129 @@
+
+        public OneTimePasswordRepository(PostgreSQLConnectionFactory connectionFactory)
+        {
+            _connectionFactory = connectionFactory;
+        }
+
+        public async Task<bool> CheckIfCodeExistsAsync(string code)
+        {
+            const string sql = "SELECT code_id FROM one_time_password WHERE code = @Code";
+            using var connection = _connectionFactory.Create();
+            connection.Open();
+
+            var result = await connection.QuerySingleOrDefaultAsync<int?>(sql, new { Code = code });
+
+            if(result == null)
+                return false;
+            else
+                return true;
+        }
+
+        public async Task<bool> CheckIfCodeExpiredAsync(string code)
+        {
+            const string sql = "SELECT expires_on FROM one_time_password WHERE code = @Code";
+            using var connection = _connectionFactory.Create();
+            connection.Open();
+
+            var result = await connection.QuerySingleOrDefaultAsync<DateTime?>(sql, new { Code = code });
+
+            // We assume if none found it is an expired code
+            if (result == null)
+                return true;
+
+            return result <= DateTime.UtcNow;
+        }
+
+        public async Task<int> DeleteAsync(string code)
+        {
+            const string sql = "DELETE FROM one_time_password WHERE code = @Code";
+
+            using var connection = _connectionFactory.Create();
+            connection.Open();
+
+            var rowsAffected = await connection.ExecuteAsync(sql, new { Code = code });
+
+            return rowsAffected;
+        }
+
+        public async Task<OneTimePassword?> GetFromCodeAsync(string code)
+        {
+            const string sql = "SELECT * FROM one_time_password WHERE code = @Code";
+
+            using var connection = _connectionFactory.Create();
+            connection.Open();
+            var result = await connection.QuerySingleOrDefaultAsync<OneTimePassword>(sql, new { Code = code });
+            return result;
+        }
+
+
+        public async Task<int> InsertAsync(OneTimePassword createOTP)
+        {
+            var sql = "INSERT INTO one_time_password (generated_by, code, created_on, expires_on, used, used_by_user, child_user) VALUES (@GeneratedBy, @Code, @CreatedOn, @ExpiresOn, @Used, @UsedByUser, @ChildUser) RETURNING code_id";
+
+            using (var connection = _connectionFactory.Create())
+            {
+                connection.Open();
+                var result = await connection.QuerySingleAsync<int>(sql, new
+                {
+                    GeneratedBy = createOTP.GeneratedBy,
+                    Code = createOTP.Code,
+                    CreatedOn = createOTP.CreatedOn,
+                    ExpiresOn = createOTP.ExpiresOn,
+                    Used = createOTP.Used,
+                    UsedByUser = createOTP.UsedByUser,
+                    ChildUser = createOTP.ChildUser
+                });
```

Comment:
Remember to close the connection when you are done with it:
```suggestion
                });
                connection.Close();
```